cs3api4lab/tests/test_cs3_file_api.py

```diff
@@ -39,194 +37,134 @@ def test_read_file(self):
-            content += chunk.decode('utf-8')
-
-        self.storage.remove(file_path, self.endpoint)
-
-        self.assertEqual(stat_by_id['filepath'], '/reva/einstein/test_read_by_share_path.txt')
-        self.assertEqual(content, content_to_check, 'File ' + file_path + ' should contain the string: ' + content_to_check)
-
+        try:
+            self.storage.write_file(file_path, content_to_write, self.endpoint)
+            stat = self.storage.stat(file_path)
+            stat_by_id = self.storage.stat(stat['inode']['opaque_id'], stat['inode']['storage_id'])
+            content = ''
+            for chunk in self.storage.read_file(file_path, self.endpoint):
+                self.assertNotIsInstance(chunk, IOError, 'raised by storage.readfile')
+                content += chunk.decode('utf-8')
+            self.assertEqual(stat_by_id['filepath'], '/reva/einstein/test_read_by_share_path.txt')
+            self.assertEqual(content, content_to_check,
+                             'File ' + file_path + ' should contain the string: ' + content_to_check)
+        finally:
+            self.storage.remove(file_path, self.endpoint)
 
     def test_read_file_no_file(self):
-
         file_path = "/test_read_no_existing_file.txt"
         content = ''
-
         with self.assertRaises(IOError, msg='No such file or directory'):
             for chunk in self.storage.read_file(file_path, self.endpoint):
                 content += chunk.decode('utf-8')
 
     def test_write_file(self):
-
         buffer = b"Testu form cs3 Api"
         file_id = "/testfile.txt"
-
-        self.storage.write_file(file_id, buffer, self.endpoint)
-
-        stat_info = self.storage.stat(file_id, self.endpoint)
-        self.assertIsInstance(stat_info, dict)
-
-        self.storage.remove(file_id, self.endpoint)
+        try:
+            self.storage.write_file(file_id, buffer, self.endpoint)
+            stat_info = self.storage.stat(file_id, self.endpoint)
+            self.assertIsInstance(stat_info, dict)
+        finally:
+            self.storage.remove(file_id, self.endpoint)
         with self.assertRaises(IOError):
             self.storage.stat(file_id, self.endpoint)
 
     def test_write_empty_file(self):
-
         buffer = b""
         file_id = "/zero_test_file.txt"
-
-        self.storage.write_file(file_id, buffer, self.endpoint)
-
-        stat_info = self.storage.stat(file_id, self.endpoint)
-        self.assertIsInstance(stat_info, dict)
-
-        self.storage.remove(file_id, self.endpoint)
+        try:
+            self.storage.write_file(file_id, buffer, self.endpoint)
+            stat_info = self.storage.stat(file_id, self.endpoint)
+            self.assertIsInstance(stat_info, dict)
+        finally:
+            self.storage.remove(file_id, self.endpoint)
         with self.assertRaises(IOError):
             self.storage.stat(file_id, self.endpoint)
 
-    def test_write_example(self):
```

Comment:
this test could be useful if we checked the notebook contents after writing. And remove it afterwards ofc

Reply:
we have _test_get_notebook_file_ in _test_cs3apismanager_ which actually does that